eis_toolkit/exploratory_analyses/statistical_tests.py

```diff
@@ -52,27 +53,61 @@ def chi_square_test(data: pd.DataFrame, target_column: str, columns: Optional[Se
 
 
 @beartype
-def normality_test(data: pd.DataFrame) -> dict:
+def normality_test(
+    data: Union[pd.DataFrame, np.ndarray], columns: Optional[Sequence[str]] = None
+) -> Union[dict, Tuple]:
     """Compute Shapiro-Wilk test for normality on the input data.
 
-    It is assumed that the input data is normally distributed and numeric, i.e. integers or floats.
-
     Args:
-        data: Dataframe containing the input data.
+        data: Dataframe or numpy array containing the input data.
+        columns: Optional Columns to be used for testing.
 
     Returns:
         Test statistics for each variable.
 
     Raises:
-        EmptyDataFrameException: The input Dataframe is empty.
+        EmptyDataException: The input data is empty.
+        NonNumericDataException: Selected data or columns contains non-numeric data.
+        InvalidParameterValueException: Input column(s) are not in the data.
+        ExceedingSampleSizeException: Input data exceeds the maximum of 5000 samples.
     """
-    if check_empty_dataframe(data):
-        raise exceptions.EmptyDataFrameException("The input Dataframe is empty.")
-
     statistics = {}
-    for column in data.columns:
-        statistic, p_value = shapiro(data[column])
-        statistics[column] = (statistic, p_value)
+    if isinstance(data, pd.DataFrame):
+        if check_empty_dataframe(data):
+            raise exceptions.EmptyDataException("The input Dataframe is empty.")
+
+        if columns is not None:
+            invalid_columns = [column for column in columns if column not in data.columns]
+            if any(invalid_columns):
+                raise exceptions.InvalidParameterValueException(
+                    f"The following variables are not in the data: {invalid_columns}"
+                )
```

Comment:
I would use here a premade check for consistency: 

```suggestion
            if not check_columns_valid(df, columns):
                raise exceptions.InvalidColumnException("All selected columns were not found in the input DataFrame.")
```